com-api/examples/basic-consumer-producer.rs

```diff
@@ -12,23 +12,23 @@
 use com_api::*;
 use com_api_gen::*;
 
-
 // Example struct demonstrating composition with VehicleConsumer
 pub struct VehicleMonitor<R: Runtime> {
     consumer: VehicleConsumer<R>,
+    producer: VehicleOfferedProducer<R>,
 }
 
 impl<R: Runtime> VehicleMonitor<R> {
     /// Create a new VehicleMonitor with a consumer
-    pub fn new(consumer: VehicleConsumer<R>) -> Self {
-        Self { consumer }
+    pub fn new(consumer: VehicleConsumer<R>, producer: VehicleOfferedProducer<R>) -> Self {
+        Self { consumer, producer }
```

Comment:
```suggestion
        let tire_subscriber = consumer.left_tire.subscribe(3).unwrap();
        Self { consumer, producer, tire_subscriber }
```
Subscriber should be created before sending data to make sure those wont be dropped